helm/cluster-vsphere/templates/copy-secret-hook-job.yaml

```diff
@@ -0,0 +1,134 @@
+#
+#
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+  namespace: "{{ $.Release.Namespace }}"
+  annotations:
+    {{- include "copyHookAnnotations" $ | nindent 4 }}
+    "helm.sh/hook-weight": "-3"
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: "{{ include "resource.default.name" $ }}-read-secret"
+  namespace: "{{ .Values.userContext.secretRef.secretNamespace }}"
+  annotations:
+    {{- include "copyHookAnnotations" $ | nindent 4 }}
+    "helm.sh/hook-weight": "-3"
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: "{{ include "resource.default.name" $ }}-create-secret"
+  namespace: "{{ $.Release.Namespace }}"
+  annotations:
+    {{- include "copyHookAnnotations" $ | nindent 4 }}
+    "helm.sh/hook-weight": "-3"
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["create"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: "{{ include "resource.default.name" $ }}-read-secret"
+  namespace: "{{ .Values.userContext.secretRef.secretNamespace }}"
+  annotations:
+    {{- include "copyHookAnnotations" $ | nindent 4 }}
+    "helm.sh/hook-weight": "-2"
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+subjects:
+  - kind: ServiceAccount
+    name: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+    namespace: "{{ $.Release.Namespace }}"
+roleRef:
+  kind: Role
+  name: "{{ include "resource.default.name" $ }}-read-secret"
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: "{{ include "resource.default.name" $ }}-create-secret"
+  namespace: "{{ .Values.userContext.secretRef.secretNamespace }}"
+  annotations:
+    {{- include "copyHookAnnotations" $ | nindent 4 }}
+    "helm.sh/hook-weight": "-2"
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+subjects:
+  - kind: ServiceAccount
+    name: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+    namespace: "{{ $.Release.Namespace }}"
+roleRef:
+  kind: Role
+  name: "{{ include "resource.default.name" $ }}-create-secret"
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+  namespace: "{{ $.Release.Namespace }}"
+  annotations:
+    {{- include "copyHookAnnotations" $ | nindent 4 }}
```

Comment:
For the Job we have `ttlSecondsAfterFinished` so I'd remove `hook-succeeded,hook-failed`. It will be hard to debug otherwise. That's why I wasn't templating this for other hooks.